webapp/lib/app/nook/view.dart

```diff
@@ -640,6 +652,16 @@ class MessageView {
     _messageTags.children.removeWhere((e) => e.dataset["id"] == tagId);
   }
 
+  void markSelectedTag(String tagId, bool selected) {
+    _messageTags.children.forEach((e) {
+      if (e.dataset["id"] == tagId) {
+        e.classes.toggle("tag--selected", selected);
+      } else {
+        e.classes.toggle("tag--selected", false);
+      }
+    });
```

Comment:
Same suggestion as above, with the same caveat that it needs testing:
```suggestion
    _messageTags.children
        .filterWhere((Element t) => t.dataset["id"] == tagId)
        .forEach((Element t) => t.classes.toggle("tag--selected", selected));
```